### PR TITLE
Component test files use examples

### PR DESCRIPTION
**Pull request by vanitabarrett**
Resolves https://github.com/alphagov/govuk-frontend/issues/1894

## What

-  Updates component tests to consistently use the examples in the YAML files, and adds any tests for untested params
- Adds a flag to hide examples that we don't want shown in the GOVUK Frontend review app
- Adds a html param and example to the fieldset component as an alternative to `caller`

## Why
Prep work for https://github.com/alphagov/govuk-frontend/issues/1895 

**Note: I've left any examples that were previously visible in the review app as-is for now as I'm not sure what the reasoning was behind adding those particular examples as something to check. It's possible some of those examples could also be hidden with the new flag.**

### CODE REVIEW

File: src/govuk/components/radios/radios.yaml
Comment (hannalaakso) [inline src/govuk/components/radios/radios.yaml:756]: Just a small thing but I **think** this example should be called 'fieldset with attributes' to be consistent with the the other examples.

File: src/govuk/components/radios/__snapshots__/template.test.js.snap
Comment (hannalaakso) [inline src/govuk/components/radios/__snapshots__/template.test.js.snap:5]: Is `undefined-hint` the correct value here?

File: src/govuk/components/summary-list/template.test.js
Comment (vanitabarrett) [reply src/govuk/components/summary-list/template.test.js:13]: @hannalaakso I think I did this because "default example" implies the tests are only using `examples.default` to test against and more often than not they were using other custom example(s). Perhaps we need to look more widely at what we actually include in the default example, but I tried to avoid changing the existing examples too much in this piece of work

File: src/govuk/components/radios/radios.yaml
Comment (vanitabarrett) [reply src/govuk/components/radios/radios.yaml:756]: @hannalaakso I think this test was about testing whether params are passed correctly to the fieldset component - `classes`, `legend` and anything related to having a hint or error message on the input, as well as passing through attributes. I think I changed it from `fieldset with attributes` to `fieldset params` to keep it a bit more generic to try and reflect this. But that may not be perfect, I'm open to suggestions!

File: src/govuk/components/radios/__snapshots__/template.test.js.snap
Comment (vanitabarrett) [reply src/govuk/components/radios/__snapshots__/template.test.js.snap:5]: Good spot 🙂

File: src/govuk/components/summary-list/template.test.js
Comment (vanitabarrett) [reply src/govuk/components/summary-list/template.test.js:14]: 😅

File: src/govuk/components/tabs/template.test.js
Comment (vanitabarrett) [reply src/govuk/components/tabs/template.test.js:106]: Ah yep, I meant to make a note of that in the PR description. I'm not sure if YAML has the concept of `undefined` so wasn't sure how to replicate that value when moving those tests across to the YAML examples.

File: src/govuk/components/radios/radios.yaml
Comment (hannalaakso) [reply src/govuk/components/radios/radios.yaml:756]: That makes sense. Could maybe call it `fieldset params for hint and error message` but that's just a suggestion.

File: src/govuk/components/accordion/accordion.yaml
Comment (hannalaakso) [inline src/govuk/components/accordion/accordion.yaml:151]: Now that I've been working on adding the new prefix/suffix examples it strikes me that having the hidden examples at the end of the file makes it harder to scan the file to understand what examples already exist and having to move between the "visible" and "hidden" examples for the same thing which is kind of annoying. 

This might just be me - but would be interesting to hear if either of you have any thoughts on how we organise the examples in the file.

File: src/govuk/components/tabs/template.test.js
Comment (hannalaakso) [reply src/govuk/components/tabs/template.test.js:106]: Yes I think you're right about `undefined`. I'm looking at the [spec for `null`](https://yaml.org/spec/1.2/spec.html#id2803362) and it seems that you could include an empty string which is different from `null`, might be something to consider 🤷

Comment (vanitabarrett) [reply src/govuk/components/tabs/template.test.js:106]: Good idea, thanks! 👍 
Added to the [accordion component](https://github.com/alphagov/govuk-frontend/pull/1924/commits/2ae642d6e47b829c7b31f2f5c59f4fe1996c70d0#diff-0dbd49b6a0a02d378b46f0523ae266e3R178) and other components in their own commits.

File: src/govuk/components/summary-list/template.test.js
Comment (vanitabarrett) [reply src/govuk/components/summary-list/template.test.js]: Fixed https://github.com/alphagov/govuk-frontend/pull/1924/commits/9bbee1e8f7898db3526e53144f607e9b5f08158c#diff-49449ca7624713ba9851d52de423e811R14

File: src/govuk/components/skip-link/template.test.js
Comment (vanitabarrett) [reply src/govuk/components/skip-link/template.test.js:13]: Moved 👍

File: src/govuk/components/radios/__snapshots__/template.test.js.snap
Comment (vanitabarrett) [reply src/govuk/components/radios/__snapshots__/template.test.js.snap]: Fixed https://github.com/alphagov/govuk-frontend/pull/1924/commits/e50c84d004a7efec453ef7aaa4023a57f6528716#diff-3109d45dad6a798504c2feec8b758819R5

File: src/govuk/components/label/label.yaml
Comment (vanitabarrett) [reply src/govuk/components/label/label.yaml]: Fixed https://github.com/alphagov/govuk-frontend/pull/1924/commits/754b1ebc530949ba0f89d7da31fb03c954b6ff3d#diff-93e61bb01b63f888cb5b03ce6b529ecfR48

File: src/govuk/components/accordion/template.test.js
Comment (vanitabarrett) [reply src/govuk/components/accordion/template.test.js]: I've tackled the bulk of these in a new commit (hopefully!): https://github.com/alphagov/govuk-frontend/pull/1924/commits/55aadb591221c1717126b64461b059052088b3ff

File: src/govuk/components/accordion/template.test.js
Comment (vanitabarrett) [reply src/govuk/components/accordion/template.test.js]: Fixed https://github.com/alphagov/govuk-frontend/pull/1924/commits/2ae642d6e47b829c7b31f2f5c59f4fe1996c70d0#diff-a29774793dd2d3af89306ffb6a0597e3L156

File: src/govuk/components/button/template.test.js
Comment (vanitabarrett) [reply src/govuk/components/button/template.test.js]: Moved as part of https://github.com/alphagov/govuk-frontend/pull/1924/commits/d0bd57d427ce598a8ef672d5cf8bdd0ef5d355da#

File: src/govuk/components/button/template.test.js
Comment (vanitabarrett) [reply src/govuk/components/button/template.test.js]: I decided to remove the first one (the one using `examples['html with no explicit element']`). Should be included in https://github.com/alphagov/govuk-frontend/pull/1924/commits/d0bd57d427ce598a8ef672d5cf8bdd0ef5d355da#

File: src/govuk/components/fieldset/template.njk
Comment (vanitabarrett) [reply src/govuk/components/fieldset/template.njk]: Fixed https://github.com/alphagov/govuk-frontend/pull/1924/commits/ceb6767c7fa3a3836dc65a8454b54bb093cf2215

File: src/govuk/components/file-upload/template.test.js
Comment (vanitabarrett) [reply src/govuk/components/file-upload/template.test.js]: Fixed as part of https://github.com/alphagov/govuk-frontend/pull/1924/commits/4e9a88b5b8549962ae7b59a3ade9ae026b55af64

File: src/govuk/components/accordion/accordion.yaml
Comment (vanitabarrett) [reply src/govuk/components/accordion/accordion.yaml:149]: I chose to separate them out as when they were all together I found it difficult to quickly see which examples were shown in the review app and which weren't. But perhaps you'd only really need to do that when working on the fixtures, and most people would benefit from having all examples together?

Comment (hannalaakso) [reply src/govuk/components/accordion/accordion.yaml:149]: It would make sense to me to group the examples for the same thing together 👍 If they're being added to the end of the file with no groupings it might become difficult to find anything there and if you're a contributor who's looking to potentially add a new example it would be helpful to see what already exists for that variant.

Comment (hannalaakso) [reply src/govuk/components/accordion/accordion.yaml:149]: I do get that you'd need to move things in several files to do that @vanitabarrett. I don't know if @36degrees has any thoughts before you do that?

**Review by 36degrees — APPROVED** [review]
Happy that any issues I flagged previously on the components that I reviewed have been addressed – and that this can be merged once @hannalaakso has approved.

File: src/govuk/components/accordion/accordion.yaml
Comment (36degrees) [reply src/govuk/components/accordion/accordion.yaml:149]: I suspect no matter our intentions they're going to get muddled up over time anyway 😬 Which might be an argument for not keeping them together, unless we think we can enforce that separation going forward? Do you think there's a logical way to order / group the examples?

Comment (hannalaakso) [reply src/govuk/components/accordion/accordion.yaml:149]: I would have said the order in which they first appear in the test file, this seems to roughly follow the order of 
```
default
with text
with html
with classes
with attributes
```
which would hopefully make the yaml easier to follow.

But I get your point that since we won't be programmatically enforcing it why even try to stick to a particular order.

**Review by hannalaakso — APPROVED** [review]
Looks great to me too 👍

**Pull request merged**
